rotkehlchen/db/upgrades/v34_v35.py

```diff
@@ -37,17 +46,28 @@ def _create_new_tables(cursor: 'DBCursor') -> None:
     """)
 
 
-def _rename_assets_identifiers(cursor: 'DBCursor') -> None:
+def _rename_assets_identifiers(write_cursor: 'DBCursor') -> None:
     """Version 1.26 includes the migration for the global db and the references to assets
-    need to be updated also in this database"""
-    cursor.execute('SELECT identifier FROM assets')
+    need to be updated also in this database.
+    We need to update the rows instead of just deleting and inserting because the rows are
+    referenced and this triggers an update in cascade.
+    """
+    write_cursor.execute('SELECT identifier FROM assets')
     old_id_to_new = {}
-    for (identifier,) in cursor:
+    for (identifier,) in write_cursor:
         if identifier.startswith(ETHEREUM_DIRECTIVE):
             old_id_to_new[identifier] = strethaddress_to_identifier(identifier[6:])
-
+        elif identifier in OTHER_EVM_CHAINS_ASSETS:
+            old_id_to_new[identifier] = OTHER_EVM_CHAINS_ASSETS[identifier]
     sqlite_tuples = [(new_id, old_id) for old_id, new_id in old_id_to_new.items()]
-    cursor.executemany('UPDATE OR IGNORE assets SET identifier=? WHERE identifier=?', sqlite_tuples)  # noqa: E501
+    # Make sure that the new ids don't exist already in the user db. This is the first version
+    # that allows the version 3 of the schema so we need to make sure that all the assets
```

Comment:
But the problem that if I understand correctly still remains is assets table cleaning. If we just delete from assets we will get foreign key hit unless there are some preparations done beforehand.

Or am I missing something?

Reply:
but this is not the asset table in the globaldb. Is the asset table in the user db. Is that the confusion? I don't follow

Reply:
Yes, this is `assets` in the user db. And it is referenced by many other tables